pkg/spec/createconfig.go

```diff
@@ -104,6 +104,7 @@ type CreateConfig struct {
 	NetworkAlias       []string               //network-alias
 	PidMode            namespaces.PidMode     //pid
 	Pod                string                 //pod
+	PodmanPath         string
```

Comment:
It has to be Podman - the cleanup process right now needs to be `podman container cleanup` even if you are not invoking from Podman

Reply:
Ok then I really don't understand what you are doing here.

Reply:
We unconditionally add a post-run cleanup process that Conmon will execute when we generate container configs. It's necessary because otherwise nothing will clean up the container's mounts, network ns, etc. This needs to point to `podman container cleanup`.

Right now, we use argv[0] for the binary to execute - we don't really know for sure where Podman lives on the system, so this works great if Podman is the one invoking libpod. However, Brent was working on a program showing off how to use Libpod directly through the Go API. In that case, argv[0] is not Podman, does not have a `container cleanup` command, and will either throw an error and do nothing (good case) or, worse, run the program that originally created the container without error - resulting in a neverending loop of containers that recreate themselves after being stopped.

Reply:
Ok then my first command is correct I believe.  If @debarshiray builds a toolbox version of libpod then he would want to call `toolbox container cleanup`, which setting the podmanpath to toolbox, makes little sense.

Reply:
I don't think we want this to point to anything that's not Podman yet - it's very specific to `podman container checkpoint`. This isn't ideal, but for now you probably need Podman on the system to use Libpod.

Reply:
Why couldn't I write my own cleanup program?

Reply:
We specify the format of the entire command that will be run - you theoretically could, but it would need to be `$YOURPROGRAM container cleanup` right now, or it wouldn't work.

We could change that too, but that's getting a little into the weeds for now. Especially since there's really no documentation on what a cleanup process for Podman needs to do.

Reply:
Just a quick note that the future Go rewrite of Toolbox will continue to fork and exec `podman`, because the resulting increase in binary size due to vendoring libpod freaked out the CoreOS team.

Even then, thanks for looking at ways to enable wider use of libpod.